### ISSUE TITLE
Clear data on exit doesn't clear tab history on exit

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#6024]
## Description 

Clear data on exit doesn't clear tab history on exit

## Steps to Reproduce

   1. Add sites to browsing history on a tab
   2. Set data to be cleared on exit
   3. Close and relaunch the browser, tab history is retained as it is

## Actual result:

[clear_data_on_exit.zip](https://github.com/brave/brave-browser/files/3614268/clear_data_on_exit.zip)

## Expected result:
Should clear browsing data if selected to clear on exit

## Reproduces how often: 

Easy

## Brave version (brave://version info)

Brave | 0.68.138 Chromium: 77.0.3865.75 (Official Build) (64-bit)
-- |--
Revision | 201e747d032611c5f2785cae06e894cf85be7f8a-refs/branch-heads/3865@{#776}
OS | Linux

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the dev channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? N/A
- Does the issue resolve itself when disabling Brave Rewards? N/A
- Is the issue reproducible on the latest version of Chrome? N/A

**Comment by Aran-Fever** [comment]
Matter of fact, Brave doesn't listen when setting the **Clear cookies and site data when you quit Brave** slider to "on" in the settings (chrome://settings/content/cookies).  Not only are my cookies and site data still listed after exiting and reopening Brave, but that setting is set back to "off"!  Really?!  *facepalm*

**Comment by bsclifton** [comment]
possibly related to https://github.com/brave/brave-browser/issues/6119 (the issue I believe @Aran-Fever mentioned)

cc: @mkarolin

**Comment by NumDeP** [comment]
I agree with Aran and Allison here [#776](https://github.com/brave/brave-browser/issues/776), which could also be related. The issue I report there I've actually noticed carried on from brave-laptop, I'm just surprised nobody else reports the same thing.

**Comment by tomlowenthal** [comment]
@srirambv Can you clarify what this issue is? What do you mean by "Add sites to browsing history on a tab"?

**Comment by Aran-Fever** [comment]
@tomlowenthal Populate your browsing history by going to a bunch of sites to your heart's content using as many tabs to your heart's content.

**Comment by rebron** [comment]
@srirambv @Aran-Fever Can you double check if your browsing history isn't cleared because what's likely happening is "Continue where you left off" is preserving your tabs? 

If you set "On Start up" -> "Open the new tab page" are you seeing expected behavior?

**Comment by bsclifton** [comment]
Any tabs that are open will be preserved in history when you relaunch the browser if `Continue where you left off` is checked (default setting)

**Comment by Aran-Fever** [comment]
Hello all.  I appreciate the recent interest for clarification and suggestions regarding this matter (and I don't mean to sound ungrateful) but I do know how to set Brave's startup behavior; I don't need to verify any settings.  Also, shortly after I found this thread (October last year), this issue was resolved by an update.  Brave properly deletes all history and cookies when set so, and keeps it set so when relaunched.  I'm not sure this would still need to be investigated unless I'm missing something, but it all seems fine to me.
